### DISCUSSION TITLE
Enabling SSL for MQTT and REST Device Services

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#138] [General]
Hi Team,

I am working on MQTT and REST device services. The requirement is I have to communicate to end device using these device services securely using certificate based authentication.

I see that currently MQTT device service is supporting username and password based authentication. And REST device service is not supporting any authentication method.

Please help me regarding how can I use these device services to communicate to end device using certificate based authentication?

Thanks,
Sudhamani

**Comment by cloudxxx8** [comment]
As you described, they don't support connections with certificates right now, but it's not too hard to add this implementation.
We just need some vulenteers to work on it.
You can file issues for your feature request first.  This is a reasonable request.

**Comment by lenny-goodell** [comment]
@sudhamani-hcl , please added enhancement request issues to each service for this.

**Comment by bnevis-i** [comment]
@sudhamani-hcl Also please be specific in the issues and be clear on where the existing functionality ends and how you want it extended.  For example, with EdgeX 3.0, device service's REST interfaces now support JWT auth, but do not support TLS listeners.  We can connect to TLS-secured message busses, but can't _______.  Et cetera.

**Comment by sudhamani-hcl** [comment]
Thank you @cloudxxx8 @lenny-intel @bnevis-i .
I will raise issue requests as suggested. It would be good if somebody implements the same.

**Comment by lenny-goodell** [comment]
Issues created, THX!. Closing discussion.
